frontend/test/jest-setup-env.js

```diff
@@ -1,19 +1,31 @@
 import "@testing-library/jest-dom";
 import fetchMock from "fetch-mock";
 
+// Fail tests on any unmocked fetch requests
+// errors in fetchMock.catch don't fail the test immediately, so we need to
+// collect them and fail the test in afterEach
+let unmockedRouteErrors = [];
+
 beforeEach(() => {
+  unmockedRouteErrors = [];
   fetchMock.mockGlobal();
+  fetchMock.catch((request) => {
+    const url = request.url || request;
```

Comment:
In what situation would `request.url` not present? I thought `request` would always be an object, then the first `request.url` would always return something.

Reply:
You're right! That was leftover from hacking at it, tidied it now!